playback/src/audio_backend/alsa.rs

```diff
@@ -1,14 +1,63 @@
 use super::{Open, Sink};
-use alsa::{Access, Format, Mode, Stream, PCM};
+use alsa::device_name::HintIter;
+use alsa::pcm::{Access, Format, HwParams, PCM};
+use alsa::{Direction, Error, ValueOr};
+use std::env;
+use std::ffi::CString;
 use std::io;
+use std::process::exit;
 
 pub struct AlsaSink(Option<PCM>, String);
 
+fn list_outputs() {
+    for t in &["pcm", "ctl", "hwdep"] {
+        println!("{} devices:", t);
+        let i = HintIter::new(None, &*CString::new(*t).unwrap()).unwrap();
+        for a in i {
+            if let Some(Direction::Playback) = a.direction {
+                println!("{:#?}", a)
```

Comment:
The output of this is pretty ugly and too verbose.
```
Hint {
    name: Some(
        "surround21:CARD=PCH,DEV=0"
    ),
    desc: Some(
        "HDA Intel PCH, ALC3220 Analog\n2.1 Surround output to Front and Subwoofer speakers"
    ),
    direction: Some(
        Playback
    )
}
```
Something closer to the `aplay -L` output would be much nicer:
```
surround21:CARD=PCH,DEV=0
    HDA Intel PCH, ALC3220 Analog
    2.1 Surround output to Front and Subwoofer speakers
```
So something like `"%s\n\t%s", a.name, a.description`

Reply:
Also, the getopts help string should be updated to "Audio device to use. Use '?' to list options if using portaudio **or alsa**"

Reply:
It still comes out a bit ugly cause the newlines in the descriptions (which they all have) don't get tabbed out properly. e.g.
```
surround21:CARD=PCH,DEV=0
	HDA Intel PCH, ALC3220 Analog
2.1 Surround output to Front and Subwoofer speakers

```
Hardly the end of the world but it's a bit shoddy. I guess either remove our tab or do a proper s/\n/\n\tb/ type string substitution on the description?